### PR TITLE
Add More tests

### PR DESCRIPTION
**Pull request by btaidm**
Adds tests for the additional load functions.
Contains commits from #100

### CODE REVIEW

**Comment by btaidm** [comment]
Ok, so I am glad it wasn't my multi-constructor code from earlier. But I am confused on why running `load_all` fails on `empty_scaler.yml` reading from the read string on CI only.
It would have to something weird with IOBuffer.

File: test/runtests.jl [hunk 1]
```diff
@@ -137,12 +137,29 @@ end
...
+        yamlString,
+        more_constructors
+    )
+    @test equivalent(stringData, expected)
+
+    allData = YAML.load_all_file(joinpath(testdir, string(test, ".data")), more_constructors)
+    @test equivalent(first(allData), expected)
+
+    allStringData = YAML.load_all(yamlString, more_constructors)
+    @test equivalent(first(allStringData), expected)
```
Comment (btaidm) [inline test/runtests.jl:161]: It fails here on `empty_scalar`. Very confused.

**Comment by kescobo** [comment]
Thanks for digging into it! Definitely weird - here's the first bit of the stacktrace (I'm sure you've seen it, just pulling it out for others)

```
Load All from String: Error During Test at D:\a\YAML.jl\YAML.jl\test\runtests.jl:168
  Test threw exception
  Expression: equivalent(first(allStringData), expected)
  AssertionError: typeof(forward!(composer.input)) == DocumentStartEvent
  Stacktrace:
   [1] compose_document(::YAML.Composer) at D:\a\YAML.jl\YAML.jl\src\composer.jl:49
   [2] compose(::YAML.EventStream) at D:\a\YAML.jl\YAML.jl\src\composer.jl:38
```

Have you pulled out and looked at what `first(allStringData)` is and compared it with `expected`? I understand it's tough to debug given that it's not happening locally, but even throwing in some print statements might help

**Comment by btaidm** [comment]
@kescobo So after putting in debug statements and trying some data massaging, the `empty_scalar` doesn't like a new line at the end of the string. I am not sure why this is, but it is a place to start.

**Comment by kescobo** [comment]
Windows often uses different line endings (`\r` I think) - not sure why it would only happen on the ci machine, but I'd be content to chomp the line endings and just compare the line contents if that's possible for this test (on phone so can't easily check). I'm pretty sure julia's `rstrip` can do this.

**Comment by btaidm** [comment]
Chomped the strings. Everything passes. Yay! :grin:

**Comment by kescobo** [comment]
Thank you! I'll try to review and merge tomorrow

**Review by kescobo — APPROVED** [review]
Looks great to me - I'll open an issue about this line ending problem - should be possible to do platform-dependent tests so that we don't have to chomp line endings, but this is good for now I think

**Pull request merged**